yarn-project/circuits.js/src/structs/header.ts

```diff
@@ -11,6 +11,8 @@ import { StateReference } from './state_reference.js';
 
 /** A header of an L2 block. */
 export class Header {
+  private blockHash: Fr | undefined;
```

Comment:
need to see if the tests play with this, im not sure if this structure is immutable